private/bufpkg/bufconnect/interceptors.go

```diff
@@ -42,65 +43,136 @@ func NewSetCLIVersionInterceptor(version string) connect.UnaryInterceptorFunc {
 	return interceptor
 }
 
+// TokenFinder finds the token for NewAuthorizationInterceptorProvider
+type TokenFinder interface {
+	// RemoteToken returns the remote token from the remote address.
+	// setFromEnvVar is true if the returned token is from the tokenEnvKey environment variable.
+	RemoteToken(address string) (token string, setFromEnvVar bool)
```

Comment:
Since the `setFromEnvVar` variable is static, you could split this out into another function to make the main call cleaner. This should also likely be named `TokenProvider` to be consistent:

```go
type TokenProvider interface {
  RemoteToken(address string) string
  IsFromEnvVar() bool
}
```